File: test_resources/rust/struct_with_external_type_models.rs
#[derive(Debug, Clone, PartialEq)]
pub struct Test {
    touch_x: f32,
    touch_y: f32,
    touch_status: TouchStatus,
}

impl Default for Test {
    fn default() -> Self {
        Self {
            touch_x: 0.0,
            touch_y: 0.0,
            touch_status: TouchStatus::default(),
        }
    }
}

File: test_resources/dart/struct_with_external_type_models.dart
class Test {
  double touchX;
  double touchY;
  TouchStatus touchStatus;

  Test({
    required this.touchX,
    required this.touchY,
    required this.touchStatus,
  });

  Test.createDefault()
      : touchX = 0.0,
        touchY = 0.0,
        touchStatus = const TouchStatusBuffersFactory().createDefault();
}

class TestBuffersFactory implements BuffersFactory<Test> {
  const TestBuffersFactory();

  @override
  Test createDefault() => Test.createDefault();
}
